### DISCUSSION TITLE
Erratic behavior of set temperature

### DISCUSSION THREAD

**Discussion by cova-fe** [jmcollin78/versatile_thermostat#747] [General]
Hi all, I'm facing a weird behavior of the thermostat that I'm not sure if it can be considered a bug, it could be. However: I'm using VT to manage a over_climate TRV, with a different temperature sensor in the middle of the room and a presence sensor. it was more or less working as expected, even though only in "strong self-regulation", as lower levels were causing too high temperature in the room before reacting. Then the VT started to behave in an unpredictable way: with setpoint to 19.5, the VT card suddenly jumped to 25 or more and the heater was fully opened. Even when I manually brought back the set temperature (using the circle) to lower levels, after some seconds the level jumped up again to seemingly random levels. I was not able to find a reason and the behavior was finally fixed by deleting the VT instance and recreating it from scratch. I'm not sure it is related, but I changed the external temperature sensor from a weather sensor to a real external thermometer, but the readings where not wildly different. Could it be that VT is sensible to some dirty data saved somewhere (as deleting the instance to fix the issue could suggest)? Did anyone experience similar behaviors? Right now it is working fine, but if it happens again I would like to collect debug data, is it enough to collect the debug log on the web UI or something else could be useful as well?

**Comment by jmcollin78** [comment]
Hello,

If you have check the "use internal device temp" (cf. https://github.com/jmcollin78/versatile_thermostat/blob/main/documentation/en/over-climate.md#compensating-for-the-internal-temperature-of-the-underlying-equipment) please uncheck it and just keep the Strong regulation.

The dirty data you are talking about is likely the internal temp of the device which can change quickly.

> though only in "strong self-regulation", as lower levels were causing too high temperature in the room

Regulation doesn't cause high temperature but more a "can reach the target more quickly".

**Reply by cova-fe** [reply]
Will try, thanks. But could this explain why the temperature shown by VT card (the one for VT, not the card showing the TRV values) was jumping from 19.5 (the programmed one) to 24 or even more? I would have expected increase in the value sent to the TRV, not in one displayed by the VTcard as set point. (in the image below, 19.5 was replaced by 24, for example). Interesting enough, the preset value indicator was turned off, basically like if I increased manually the setpoint using the "circle" or +- buttons. When it happened I lowered manually the setpoint on the card, but immediately after it jumped again to 24. (I use 24 as example, but it was not always the same). I had no presets with 24 as temperature, btw. Basically it seemed like if an external automation or agent was changing the settings in the card/thermostat rather than a weird regulation in the underpinning algorithm to set the TRV value.
![image](https://github.com/user-attachments/assets/38bad237-f78e-46f2-8682-958a6f776d2c)

**Reply by jmcollin78** [reply]
In the card you can display the regulated temp (the temperature sent by VTherm to the device).
